tests/test_job.py

```diff
@@ -483,8 +522,108 @@ def test_job_14_update_parameters_from_file(self):
+    def test_job_16_upload_agg_optimizer_aux_var(
+        self,
+        patch_serializer_dump,
+        patch_repository_upload,
+        patch_uuid,
+    ):
+        """Test 'upload_agg_optimizer_aux_var' with both shared and node-specific info."""
+        # Set up: two target nodes; aux vars with shared and node-specific info.
+        self.job.nodes = ["node-1", "node-2"]
+        setattr(self.job, "_keep_files_dir", "dir")  # simplify dump paths
+        aux_var = {
+            "module_a": {"key": "val"},
+            "module_b": {f"node-{i + 1}": {"key": "val"} for i in range(3)},
+        }
+        patch_uuid.return_value = "uuid"
+        patch_repository_upload.return_value = {"file": "url"}
+        # Call the tested method.
+        url_shared, url_bynode = self.job.upload_agg_optimizer_aux_var(aux_var)
+        # Verify that results and mock calls match expectations.
+        self.assertEqual(url_shared, "url")
+        self.assertDictEqual(url_bynode, {"node-1": "url", "node-2": "url"})
+        patch_serializer_dump.assert_has_calls([
+            call({"module_a": {"key": "val"}}, "dir/aux_var_shared_uuid.mpk"),
+            call({"module_b": {"key": "val"}}, "dir/aux_var_node_node-1_uuid.mpk"),
+            call({"module_b": {"key": "val"}}, "dir/aux_var_node_node-2_uuid.mpk"),
+        ], any_order=True)
+        patch_repository_upload.assert_has_calls([
+            "dir/aux_var_shared_uuid.mpk",
+            "dir/aux_var_node_node-1_uuid.mpk",
+            "dir/aux_var_node_node-2_uuid.mpk",
+        ], any_order=True)
+
+    @patch('uuid.uuid4', autospec=True)
+    @patch('fedbiomed.common.repository.Repository.upload_file', autospec=True)
+    @patch('fedbiomed.common.serializer.Serializer.dump', autospec=True)
+    def test_job_17_upload_agg_optimizer_aux_var_shared_only(
+        self,
+        patch_serializer_dump,
+        patch_repository_upload,
+        patch_uuid,
+    ):
+        """Test 'upload_agg_optimizer_aux_var' with shared info only."""
+        # Set up: two target nodes; aux vars with shared and node-specific info.
+        self.job.nodes = ["node-1", "node-2"]
+        setattr(self.job, "_keep_files_dir", "dir")  # simplify dump paths
+        aux_var = {
+            "module_a": {"key": "val"},
+            "module_b": {"key": "val"},
+        }
+        patch_uuid.return_value = "uuid"
+        patch_repository_upload.return_value = {"file": "url"}
+        # Call the tested method.
+        url_shared, url_bynode = self.job.upload_agg_optimizer_aux_var(aux_var)
+        # Verify that results and mock calls match expectations.
+        self.assertEqual(url_shared, "url")
+        self.assertDictEqual(url_bynode, {})
+        patch_serializer_dump.assert_called_once_with(
+            aux_var, "dir/aux_var_shared_uuid.mpk"
+        )
+        patch_repository_upload.assert_called_once_with(
+            "dir/aux_var_shared_uuid.mpk"
+        )
```

Comment:
Just a (very) minor remark: we could assign `"url"` to a variable, making the test less bug prone: I mean:
```
url_link = "url"
...
self.assertEqual(url_shared, url_link)
```
same for `"dir"`

Reply:
Indeed; I corrected that. Thanks!